docs/best-practices.md

```diff
@@ -0,0 +1,41 @@
+### How to select good environmental variable name?
+When selecting for environmental variable name, please check Dotnet cli convention examples.
+Please check discussion [here](https://github.com/NuGet/Client.Engineering/issues/882#issuecomment-816118683) for more details, please note this's not ultimate guide just our learning from last experience and hoping to help you, feel free to improve it. 
```

Comment:
Unfortunately this is not very helpful given that it's private. 

I'd suggest summarizing the best learnings out of that.